broker-core/src/test/java/io/zeebe/broker/workflow/WorkflowInstanceFunctionalTest.java

```diff
@@ -351,4 +355,119 @@ public void shouldReprocessWorkflowInstanceRecordsWhenWorkflowIsTemporarilyUnava
     final long newWorkflowInstancekey = testClient.createWorkflowInstance(PROCESS_ID);
     assertThat(newWorkflowInstancekey).isGreaterThan(0);
   }
+
+  @Test
+  public void shouldHaveCorrectBpmnElementTypes() {
+    // given
+    final BpmnModelInstance model =
+        Bpmn.createExecutableProcess(PROCESS_ID)
+            .startEvent("start")
+            .parallelGateway("split")
+            .serviceTask("task", t -> t.zeebeTaskType("task"))
+            .parallelGateway("merge")
+            .endEvent("end")
+            .moveToNode("split")
+            .intermediateCatchEvent("timer")
+            .timerWithDuration("PT1S")
+            .connectTo("merge")
+            .done();
+
+    brokerRule.getClock().pinCurrentTime();
+    testClient.deploy(model);
+
+    // when
+    testClient.createWorkflowInstance(PROCESS_ID);
+
+    assertThat(RecordingExporter.jobRecords(JobIntent.CREATED).exists()).isTrue();
```

Comment:
You don't need to check this here. It is already checked in `completeJobOfType()`.